docs/proposals/fp2.rst

```diff
@@ -0,0 +1,145 @@
+FP2: Local Lint Rules
+=====================
+
+By default, Fixit expects rules to be referenced by their fully-qualified
+module and/or class name. For example, ``fixit.rules`` refers to the bundled set
+if lint rules shipped with Fixit. Third party rules available in the environment
+can similarly be referenced by their module name, as long as they are installed
+and importable by Fixit at runtime. These will be collectively referred to as
+"global rules" here for sake of clarity.
+
+However, in many cases, it is also useful for projects using Fixit to build
+custom lint rules specific to the project being linted. It is beneficial for
+those rules to be defined within the codebase being linted, without needing to
+first install those lint rules into the environment before running Fixit.
+
+These are defined as "local rules", and will require dedicated behavior from
+Fixit to discover and import these rules at runtime, as well as special
+configuration syntax for enabling these local rules in a project.
+
+
+Configuration
+-------------
+
+When configuring the set of enabled or disabled rules, any local rules must
+be marked with a single leading period, and referred to using their path
+relative to the configuration file's directory.
+
+For example, a configuration file at ``project/fixit.toml`` could include
+rules defined in ``project/some/local/rules.py`` with the following:
+
+.. code-block:: toml
+
+    [tool.fixit]
+    enable = [".some.local.rules"]
+
+
+References to local rules should be accepted in either ``enable`` or ``disable``
+options, including overrides, to provide the same selection criteria available
+to global rules:
+
+.. code-block:: toml
+
+    [tool.fixit]
+    enable = [".rules"]
+
+    [[tool.fixit.overrides]]
+    path = "project1"
+    disable = [".rules.PickyRule"]
```

Comment:
Additionally, this is from FP1, but why not `.rules:PickyRule` with a colon to be more like PEP 517? If that's not the original location where PickyRule is defined, can I assume it does a lookup of `PickyRule.__module__` to determine a canonical path?  Is it an error to disable something that fixit doesn't think is enabled?

Reply:
I think I would personally prefer to use `foo.bar:Baz`, but `foo.bar.Baz` is what Fixit has used for a while, and basically just tries to import `foo`, then walk the dotted parts to find the final result.

If we switch to `foo.bar:Baz`, then we could just assume that `foo.bar` will always be the full module name, and import that directly rather than needing to walk modules/objects and assume/hope that everything is imported appropriately.

@zsol, since we have the opportunity to change this, do you have an opinion?

>  Is it an error to disable something that fixit doesn't think is enabled?

No, it just goes in the set of things to disable, and if that thing never got enabled, then nbd.

Reply:
I'm happy to do `foo.bar:Baz`, it's both more clear to the reader and simpler to implement.